include/ginkgo/core/base/range.hpp

```diff
@@ -155,6 +166,13 @@ GKO_ATTRIBUTES GKO_INLINE constexpr bool operator!=(const span &first,
 }
 
 
+GKO_ATTRIBUTES GKO_INLINE constexpr span operator+(const span &first,
+                                                   const span &second)
+{
+    return gko::span{first.begin + second.begin, first.end + second.end};
+}
+
+
```

Comment:
Could you make this a `friend` inside `span` instead? That prevents accidental implicit conversions when looking for suitable `operator+` overloads. Also, what use case do you have in mind for this operation? span + index I can see, but span + span looks a bit unusual to me. Maybe make this a named operation instead?

Reply:
Mainly when combining two contiguous spans. Yeah, I think it might make sense to make this a named operation.

Reply:
Combining two contiguous spans to me sounds like [a,b) + [b,c) = [a,c), but that is not what this operation is intended for, right?

Reply:
Yes, good catch!  I confused this with my usage of separate usages of row_spans and column_spans. `[a,b) + [b,c) = [a,c)` is the right way to go. I will update this.

Reply:
But that operation is only defined if the end of one range does not match the beginning of the next, so I would be really careful about adding something like this.

Reply:
Yeah, I agree. I removed it for now.